docs/technical/integration-testing-strategy.md

```diff
@@ -0,0 +1,96 @@
+
+### Classify a message
+
+This test asserts that when DMR receives an unclassified message which is correctly formed, it is able to classify it and reply to the calling participant with the classification. This is steps 1-4 from [Milestones.md](https://github.com/buerokratt/Project-Documentation-Management/blob/main/Milestones.md) and represents milestone 1.
+
+Input: Test requests that mock bot generates an unclassified message with all appropriate headers and submits it to DMR.
+
+Expected output: DMR calls back to the mock bot with a classification. Assert that the classification is what was expected for the message contents.
+
+### Bot-to-bot
+
+This test asserts that a bot can route a classified message to another bot via DMR. This represents steps 6 and 7 from [Milestones.md](https://github.com/buerokratt/Project-Documentation-Management/blob/main/Milestones.md) and milestone 2
+
+Input: Test requests that mock bot generates a classified message with all appropriate headers and submits it to DMR.
+
+Expected output: DMR forwards the message to the correct mock bot. Test asserts that the message was received and is in the expected format.
+
+## How to make assertions on HTTP calls
+
+The architecture of the Bürokratt system operates on a series of chained HTTP requests and the final step of any interaction is for a mock bot to receive a HTTP request from DMR.
+
+In order to assert that these HTTP requests are being received, we can use some of the existing endpoints that are included in the mock bot to create chats and post messages to a chat (thereby sending it to DMR).
+
+There may be some minor extensions required to mock bot to ensure that when DMR responds back to the bot, the message is added to the Chat (currently it is disposed). This will enable us to simply get the chat and inspect the contents which should contain both the initial message out to DMR and the classified response back from DMR.
+
+## Platform and Tooling
+
+All integration tests will be written using C# and XUnit. This is a consistent choice with the unit tests for the Bürokratt system.
+
+The tests will use an `appsettings.json` and `appsettings.developments.json` files to store configuration information for accessing components (endpoints, api keys etc).
+
+Developers will be able to run tests locally using the `dotnet test` command.
+
+## Pipelines
+
+The goal of running tests as part of pipelines that the outcome of a test can be used a gate for progressing code to the next stage. For example if a test fails, then the automated progression of code from dev to production can be halted.
+
+The challenge with running integration tests as part of a pipeline is that the code must be available in its integrated form in order for the test to work, so to a certain extent the pipline must have progressed before integration tests can run.
```

Comment:
This was a little difficult to read so here my suggestion on how to reword it
```suggestion
The challenge with running integration tests as part of a pipeline is that the code must be deployed to the `dev` environment (our first integrated environment), meaning that it needs to have merged into the `main` branch already.
```

Reply:
I'm not sure if the suggestion is quite right either because we can run the tests with Docker Compose. I'll have a look and come up with some better words.

Reply:
I ended up removing the paragraph entirely as it was not really adding anything